### ISSUE TITLE
Writing a Symbol Blog article on XYMbassadors experience learning Python

### ISSUE THREAD

**Issue by simonm99** [symbol/xymbassadors#45]
I am thinking of writing a blog post about your collective experience of learning Python. I think it would be a really interesting read and it might get others thinking about taking the plunge and doing the same. I am not sure what the article would look like yet but I am thinking topics could include:

- Things you enjoyed
- Things you didn't enjoy!
- What was the most difficult part
- What kept you motivated
- What resources you used
- Have you experimented writing your own code to solve a problem (outside of the set examples)
- Any top tips that you would give to someone who has just decided to learn Python
- What you plan to do with your new found skills

I am not sure how I will put this together as there are a lot of XYMbassadors learning Python but maybe a start would be me compiling a list of questions for you and asking you to fill them in then I will try to shape it into an article.

**Comment by C2daB** [comment]
I’ve enjoyed learning Python and it was a great experience. 
I was already familiar with other coding languages, so it didn’t take me too much to get through Python I – V on [Progate](https://progate.com/courses/python). 

There is a lot more to learn and we shouldn’t stop here.

Now I’m using [PyCharm](https://www.jetbrains.com/pycharm/) community edition to learn more and start writing my own code. 
You can install different packages to make your code smoother and ease your workflow. 

At the moment I’m in a searching process, going through Google pages that offer information, python plugins, and examples of code I could use in my own code. 

In this process I found https://earsketch.gatech.edu/earsketch2 
Go to https://earsketch.gatech.edu/earsketch2/?sharing=mGnNo3vaZlNJ_5mbOLvEGQ  press RUN and wait for the code to load in your DAW, then press the Play button. You will see that you can make music by only using Python code and a library of WAV samples.

This is one way to make music using Python code, the easiest way I could find so far, but I’m looking for a more complex open-source plugin that I could use to build an app alike on the Symbol blockchain.

So wish me luck, cause I need it. 😄

**Comment by AngelNEM** [comment]
Since childhood, I wanted to study programming, physics at school and university was my favorite subject, but life turned out in such a way that I had to go to work in the field of culture and sports!  
Then there was the fear that programming is very difficult, and in order to master it, you need to spend several years studying at the university, studying computer and mathematics inside and out.  
But as it turned out, thanks to our team Xymambassador is not so scary, and I want to tell you about it.
 To begin with, I want to emphasize an important idea for beginners - in order to learn a programming language, you do not need to have special knowledge.
 If you don't understand a topic, try studying it in different sources.
It is not necessary to buy courses, you can get the necessary skills from books, the main thing is to approach the choice thoroughly
 Be sure to practice the knowledge gained
 When learning from books, choose the latest publications with up-to-date information
 The most important thing a programmer needs is a passion for programming, everything else is secondary.
 I liked Python because it can do any job.  It has a library for mobile applications, for desktop applications, you can write automation scripts for servers on it, you can use it for web applications.
 The syntax turned out to be really simple.  How to write text in English.  
I learn Python from YouTube videos, articles!  To keep the courses from becoming useless, only one thing will help - practice. If you just watch and listen, it won't work.  For a beginner, it will even be a practice to rewrite the code that the instructor shows on the screen.  Rewrite and think about how it works. The required minimum for classes is four hours a week. 
Two days a week for two hours.  
But I would recommend setting aside an hour a day to write your own code.  
The main thing is not to dive into programming with your head, so that there is no riot of magic.  When you study all three days, you have a mess in your head!  Very difficult!  There are about 70 functions and several dozen reserved words in a basic Python configuration, but even a cool programmer doesn't necessarily use all of them.  
That is, in order to learn a hundred words and understand what they are doing, you can spend from one to three weeks, if you want, and work actively. 
Now I'm slowly putting together a very simple text game piece by piece, https: // github.com/kitao helps me with this /pyxel/blob/master/README.ja.md
But when faced with the first real problem, I threw everything away.  But programming problems are inevitable;  constantly something does not work, and it is completely incomprehensible why. I banged my head against the keyboard and tried to figure out incomprehensible error messages.  Moreover, my ultimate goal was to create a game that I would not care about and that no one would ever play.  
I would like to create a simple game and bind it to Symbol transactions!  I will gradually come to this!  It's important to stay motivated!  Inspiration ! When you are learning something difficult (especially if you are a beginner) you want small wins as early as possible.  They help you believe that you can actually do what you are about to do.  I was inspired by the work of my friends MIkun, Klim, DJ СB, Simon, they always inspire and help! 
If something is not clear to me, I always turn to my friends and they help me - it is always customary in our team to help and motivate each other!  we are NEMber - and we always help each other!🔥✊🏻💜⛓
I learned that this is an ambitious long-term goal that may not be achievable: even the best Python developers don't know it very well.  Learn Python to understand and write simple code!  Be educated and versatile!  
I can recommend the book “Paul Bury".  
Learning Python Programming and the Tkinter GUI (better to go to pygame or pyqt5 first!
   Better to learn the basics first.  And there will always be time for your own projects.  Do not lose faith, go to your goal, imagining how long it will take to learn Python, (a similar feat), I often wanted to give up everything.  
I got a lot of information - the official manual, blog articles, videos.  
I studied all this diligently, noticing individual constructions of language and technology, but there was frankly little sense.  
Take a break, awareness of your goals will give you strength and movement forward!  I wish you success and never give up!  
Regards, your AngelNEM ✊🏻💜
P.s sorry for my english, if that ☺️![image](https://user-images.githubusercontent.com/87577525/130372497-7ad3058b-9933-4d30-8c15-58660043acd1.png)

**Comment by HexagonTR** [comment]
<img width="1457" alt="Screen Shot 2021-08-23 at 22 09 11" src="https://user-images.githubusercontent.com/82543746/130507845-b28a40d5-78a7-45d9-aca5-657371d9c233.png">
We feel more and more every day that learning to code is now a necessity. The Symbol community is like a school, and I said "now it's time to start learning coding" because of the developer team saying that coding is the future and we need to learn.
Then I gave it as far as Python is the most suitable coding language for learning coding. My goal was to produce an implement that works on the web. That's why I became interested in Flask along with Python. This duo has a lot of documents and videos on the internet. Currently, as a learning method, I act with the method of operating by making small changes in the existing codes and seeing the differences. My main goal is to solve algorithm logic. Code writing rules are already being corrected by IDEs or you can easily find the right version on the internet.

Turning the coding into an algorithm and thinking about the code as a whole... This seems to take a very long time. :) But make sure that you make the process fun by making fun projects and listening to music, it works.

In summary, in order to turn your ideas into reality or to communicate better with a developer, you need to know entry-level coding languages ​​and I am currently focusing on this stage.

By the way, running symbol nodes on different platforms is one of my biggest hobbies. Just like they do in Doom :) https://www.gamesradar.com/12-things-that-prove-that-doom-will-run-on-literally- anything/

The following node is running on windows 10. You can use it for harvest if you want.

http://symbolnode.hexagontr.keenetic.link:3000

Best wishes,

HexagonTR

**Comment by klimgeran** [comment]
### I started my learning of the Python programming language with the #Progate course

https://twitter.com/GeranKlim/status/1423540683126689793
![E8FutC2UcAEwuXs](https://user-images.githubusercontent.com/30126937/130584568-b6648f52-a272-4a2a-ae98-560882bc36a8.jpeg)

Then I decided to make a small game and made the calculation of points in the form of XYM, adding 10 XYM, for each cube when the snake eats it, then I changed the points by 1 XYM.
https://twitter.com/GeranKlim/status/1427539604308070405

<img width="354" alt="Screen Shot 2021-08-24 at 4 34 49 PM" src="https://user-images.githubusercontent.com/30126937/130586730-c780f24e-0549-47fe-949e-77efd3c77d44.png">
<img width="355" alt="Screen Shot 2021-08-24 at 4 34 56 PM" src="https://user-images.githubusercontent.com/30126937/130586742-46d11150-86f8-4479-9b07-bce4da9d5a41.png">

https://github.com/klimgeran/Symbol_Game - The code itself is here. 
<img width="854" alt="Screen Shot 2021-08-24 at 4 47 43 PM" src="https://user-images.githubusercontent.com/30126937/130586900-a67dfc9e-04ec-4549-b3f7-5d19eb35c35e.png">

Here, you can see how to send transactions using Python on the Symbol blockchain:

https://twitter.com/GeranKlim/status/1428392002735857670


https://user-images.githubusercontent.com/30126937/130587805-01ee584f-e11d-404b-8949-1fc5466d6fe8.MOV

This is how my training goes...




_______________________________________

Connect to my #Symbol node 👉
: http://node.symbolplatform.ru:3000/

**Comment by johnnytongki** [comment]
Thanks to our team, the community manager has a strong desire to learn #Python.

Reading lessons and completing assignments, I got a good understanding of the syntax of the language. But - there was no feeling that I really own it. The first rakes I stepped on were tabs and spaces. I re-read the written code a bunch of times and did not understand why it did not work, I tried to change the logic, tested and debugged. It turned out that the whole point was that I was interfering with tabs and spaces.

Now I have some idea of ​​how programming languages ​​work, what they are. English! It is 100% needed, it is difficult and not comfortable to switch to it, but it needs to be done. I'm still in the process, studying, the Progate course - I didn't really like it.

 https://twitter.com/tongkijohnny/status/1424260376384839684?s=21

<img width="1129" alt="Python" src="https://user-images.githubusercontent.com/87557879/130638221-4d6d5c56-8e41-4d89-b595-8987de7ad0fa.png">
